### PR TITLE
chore: migrate to main branch

### LINKED ISSUE #137: Migrate to main branch
Contributing to googleapis/google-cloud-python#10579, switching the default branch to main branch. Also replacing all occurrences of master to main.

### PR DESCRIPTION
**Pull request by dandhlee**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-datalabeling/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #137 🦕

### CODE REVIEW

File: samples/AUTHORING_GUIDE.md [hunk 1]
```diff
@@ -1 +1 @@
-See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md
\ No newline at end of file
+See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md
```
Comment (parthea) [inline samples/AUTHORING_GUIDE.md:1]: I get 404 with the new link

**Review by parthea — COMMENTED** [review]
Added minor observations, otherwise LGTM.

File: UPGRADING.md [hunk 2]
```diff
@@ -66,7 +66,7 @@ In `google-cloud-datalabeling<1.0.0`, parameters required by the API were positi
 
 In the 1.0.0 release, all methods have a single positional parameter `request`. Method docstrings indicate whether a parameter is required or optional.
 
-Some methods have additional keyword only parameters. The available parameters depend on the [`google.api.method_signature` annotation](https://github.com/googleapis/googleapis/blob/master/google/cloud/datalabeling/v1beta1/data_labeling_service.proto#L48) specified by the API producer.
+Some methods have additional keyword only parameters. The available parameters depend on the [`google.api.method_signature` annotation](https://github.com/googleapis/googleapis/blob/main/google/cloud/datalabeling/v1beta1/data_labeling_service.proto#L48) specified by the API producer.
```
[the remark this answers is not in the document]
Comment (dandhlee) [reply UPGRADING.md]: Done.

File: README.rst [hunk 3]
```diff
@@ -85,4 +85,4 @@ Next Steps
    APIs that we cover.
 
 .. _Data Labeling API Product documentation:  https://cloud.google.com/data-labeling/docs
-.. _repository’s main README: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/README.rst
+.. _repository’s main README: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/main/README.rst
```
[the remark this answers is not in the document]
Comment (dandhlee) [reply README.rst]: Done.

File: samples/AUTHORING_GUIDE.md (hunk 1 above)
Comment (dandhlee) [reply samples/AUTHORING_GUIDE.md]: Done.

**Comment by dandhlee** [comment]
Please take a look again!

**Review by parthea — APPROVED** [review]

**Pull request merged**